File: Week 02/id_343/LeetCode_144.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func preorderTraversal(root *TreeNode) []int {
    if root == nil {
        return nil
    }
    
    l := preorderTraversal(root.Left)
    r := preorderTraversal(root.Right)
    
    return append(append([]int{root.Val}, l...), r...)
}

File: Week 02/id_113/LeetCode144.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def preorderTraversal(self, root: TreeNode) -> List[int]:
        traverse_path = []
        if root is None:
            return []

        traverse_path.append(root.val)
        traverse_path.extend(self.preorderTraversal(root.left))
        traverse_path.extend(self.preorderTraversal(root.right))

        return traverse_path
